pkg/executables/govc.go

```diff
@@ -533,6 +534,9 @@ func (g *Govc) validateAndSetupCreds() (map[string]string, error) {
 	} else if govcURL, ok := os.LookupEnv(govcURLKey); !ok || len(govcURL) <= 0 {
 		return nil, fmt.Errorf("%s is not set or is empty: %t", govcURLKey, ok)
 	}
+	if govcDatacenter, ok := os.LookupEnv(govcDatacenterKey); !ok || len(govcDatacenter) <= 0 {
```

Comment:
During e2e this will fail for vsphere tests as well since we are not setting this envar in code build. You either need to update code build job file to set it or follow pattern for other vars where we read the eksa value and set it.